File: problems/hackerrank/Tutorials/30 Days of Code/Day 3: Intro to Conditional Statements/solution.py
"""
Given an integer, n, perform the following conditional actions:
    If n is odd, print Weird
    If n is even and in the inclusive range of 2 to 5, print Not Weird
    If n is even and in the inclusive range of 6 to 20, print Weird
    If n is even and greater than 20, print Not Weird
"""

def weird(n):
    if n % 2 != 0:
        print("Weird")
    else:
        if 2 <= n and n <= 5:
            print("Not Weird")
        if 6 <= n and n <= 20:
            print("Weird")
        if 20 < n:
            print("Not Weird")
    return 0;


def main():
    N = int(input())
    weird(N);


if __name__ == '__main__':
    main()

File: problems/hackerrank/Tutorials/30 Days of Code/Day 3: Intro to Conditional Statements/solution.cpp
#include <iostream>
using namespace std;


/*
Given an integer, n, perform the following conditional actions:
    If n is odd, print Weird
    If n is even and in the inclusive range of 2 to 5, print Not Weird
    If n is even and in the inclusive range of 6 to 20, print Weird
    If n is even and greater than 20, print Not Weird
*/
int weird(int n) {
    if ((n % 2) != 0) cout << "Weird" << endl;
    else {
        if ((2 <= n) && (n <= 5)) cout << "Not Weird" << endl;
        if ((6 <= n) && (n <= 20)) cout << "Weird" << endl;
        if (20 < n) cout << "Not Weird" << endl;
    }
    return 0;
}


int main(){
    int N;
    cin >> N;
    weird(N);
    return 0;
}
